### PR TITLE
Helper to resolve proxy from URL in any environment, or using extension-API

### LINKED ISSUE #5689: Extension api for resolving a proxy for an url
Api for something like:

```ts
electron.webContents.getAllWebContents()[0]?.session.resolveProxy(url)
```

### PR DESCRIPTION
**Pull request by Iku-turso**
Note: given there are no open windows (ie. webContents), the helper will throw, and log the error.

Closes #5689

### CODE REVIEW

File: src/behaviours/proxy/resolve-proxy/main/electron.injectable.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import electron from "electron";
+
+const electronInjectable = getInjectable({
```
Comment (jakolehm) [inline src/behaviours/proxy/resolve-proxy/main/electron.injectable.ts]: `electronInjectable` sounds like something that is common for other stuff. Are we going to move this whenever there is a need?

File: src/extensions/common-api/proxy.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { asLegacyGlobalFunctionForExtensionApi } from "../as-legacy-globals-for-extension-api/as-legacy-global-function-for-extension-api";
+import { resolveProxyInjectionToken } from "../../behaviours/proxy/resolve-proxy/common/resolve-proxy-injection-token";
+
+export const resolveProxy = asLegacyGlobalFunctionForExtensionApi(resolveProxyInjectionToken);
```
Comment (jakolehm) [inline src/extensions/common-api/proxy.ts]: Is there a "non-legacy" way to export functions for extensions?

Comment (jakolehm) [reply src/extensions/common-api/proxy.ts]: `resolveProxy` as a name is very generic (would `resolveSystemProxy` make more sense?). What kind of behavior extension api user expects from this? 🤔 At least we should have api docs (tsdoc) that shortly summarizes what it is.

**Review by jakolehm — COMMENTED** [review]
Functionality looks good but I'm not sure about the api naming and the fact that we decided to use `src/behaviours` for this kind of stuff. Folder structure is something we can shuffle later but we probably need to rethink the api name _or_ change the functionality a bit.

File: src/behaviours/proxy/resolve-proxy/main/electron.injectable.ts (hunk 1 above)
Comment (Iku-turso) [reply src/behaviours/proxy/resolve-proxy/main/electron.injectable.ts]: Right you are, it is general, and will move to more shared scope when something else requires it :)

File: src/extensions/common-api/proxy.ts (hunk 2 above)
Comment (Iku-turso) [reply src/extensions/common-api/proxy.ts]: > Is there a "non-legacy" way to export functions for extensions?

All stuff exposed in current extension-API is involuntary **global state**. A completely clean function can be exported as it is, "non-legacy"... But that other stuff with side-effects and lifecycles and dependencies and all, cannot be but packaged as legacy :/

This is a considerable damage that could be fluently solved in next iteration of extension-API.

Comment (Iku-turso) [reply src/extensions/common-api/proxy.ts]: > would resolveSystemProxy make more sense?

A bit I guess. Renamed :)

Comment (Iku-turso) [reply src/extensions/common-api/proxy.ts]: > What kind of behavior extension api user expects from this? 🤔 At least we should have api docs (tsdoc) that shortly summarizes what it is.

Hmm. Electron's own documentation for `session.resolveProxy()` is pretty [terse](https://www.electronjs.org/docs/latest/api/session#sesresolveproxyurl), as it just says "Resolves with the proxy information for `url`." Duh. I suppose you already have better understanding for what this does? Share? :)

Comment (Iku-turso) [reply src/extensions/common-api/proxy.ts]: TSDoc added from conversation elsewhere.

**Review by jakolehm — APPROVED** [review]
LGTM

**Review by jansav — APPROVED** [review]

**Pull request merged**